agent/06_agent_create_cluster.sh

```diff
@@ -163,13 +163,74 @@ function mce_complete_deployment() {
   mce_apply_postinstallation_manifests ${mceManifests}
 }
 
-create_image
-if [[ "${AGENT_DISABLE_AUTOMATED:-}" == "true" ]]; then
-  disable_automated_installation
-fi
+# Generate the PXE artifacts
+function create_pxe_files() {
+    local asset_dir=${1}
+    local openshift_install=${2}
+    "${openshift_install}" --dir="${asset_dir}" --log-level=debug agent create pxe-files
+}
+
+# Setup the environment to allow iPXE booting, by reusing libvirt native features
+# to configure dnsmaq tftp server and pxe boot file. Note that this is supported
+# only for IPv4.
+function setup_pxe_server() {
+    local PXE_SERVER_DIR=${WORKING_DIR}/pxe
+    mkdir -p ${PXE_SERVER_DIR}
+
+    # Create the iPXE boot file
+    local BASEURL=tftp://${PROVISIONING_HOST_EXTERNAL_IP}
```

Comment:
Does the tftp server run internally in libvirt?

Reply:
Yes, it is provided by the dnsmasq instance used by libvirt (for each network)